server/apps/archive_broadcast/broadcast.py

```diff
@@ -0,0 +1,293 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+
+import logging
+import json
+from eve.utils import ParsedRequest
+from eve.versioning import resolve_document_version
+from flask import request
+from apps.archive.common import CUSTOM_HATEOAS, insert_into_versions, get_user, \
+    ITEM_CREATE, BROADCAST_GENRE, is_genre
+from apps.packages import TakesPackageService
+from superdesk.resource import Resource, build_custom_hateoas
+from superdesk.services import BaseService
+from superdesk.metadata.utils import item_url
+from superdesk.metadata.item import CONTENT_TYPE, CONTENT_STATE, ITEM_TYPE, ITEM_STATE
+from superdesk import get_resource_service, config
+from superdesk.errors import SuperdeskApiError
+from apps.archive.archive import SOURCE
+from apps.publish.content.common import ITEM_CORRECT, ITEM_PUBLISH
+
+
+logger = logging.getLogger(__name__)
+# field to be copied from item to broadcast item
+FIELDS_TO_COPY = ['urgency', 'priority', 'anpa_category', 'type',
+                  'subject', 'dateline', 'slugline', 'place']
+ARCHIVE_BROADCAST_NAME = 'archive_broadcast'
+
+
+class ArchiveBroadcastResource(Resource):
+    endpoint_name = ARCHIVE_BROADCAST_NAME
+    resource_title = endpoint_name
+
+    url = 'archive/<{0}:item_id>/broadcast'.format(item_url)
+    schema = {
+        'desk': Resource.rel('desks', embeddable=False, required=False, nullable=True)
+    }
+    resource_methods = ['POST']
+    item_methods = []
+    privileges = {'POST': ARCHIVE_BROADCAST_NAME}
+
+
+class ArchiveBroadcastService(BaseService):
+
+    takesService = TakesPackageService()
+
+    def create(self, docs):
+        service = get_resource_service(SOURCE)
+        item_id = request.view_args['item_id']
+        item = service.find_one(req=None, _id=item_id)
+        doc = docs[0]
+
+        self._valid_broadcast_item(item)
+
+        desk_id = doc.get('desk')
+        desk = None
+
+        if desk_id:
+            desk = get_resource_service('desks').find_one(req=None, _id=desk_id)
+
+        doc.pop('desk', None)
+        doc['task'] = {}
+        if desk:
+            doc['task']['desk'] = desk.get(config.ID_FIELD)
+            doc['task']['stage'] = desk.get('incoming_stage')
+
+        doc['task']['user'] = get_user().get('_id')
+        genre_list = get_resource_service('vocabularies').find_one(req=None, _id='genre') or {}
```

Comment:
The `VocabulariesService.find_one()` gives inactive vocabularies also. So, Inactive Vocabularies also needs to be filtered.